src/btree/bt_curprev.c

```diff
@@ -671,23 +685,45 @@ __wt_btcur_prev(WT_CURSOR_BTREE *cbt, bool truncating)
     WT_SESSION_IMPL *session;
     size_t total_skipped, skipped;
     uint32_t flags;
-    bool newpage, restart;
+    bool key_out_of_bounds, newpage, restart, need_walk;
 
     cursor = &cbt->iface;
+    key_out_of_bounds = false;
+    need_walk = false;
     session = CUR2S(cbt);
     total_skipped = 0;
 
+    /*
+     * If we have a bound set we should position our cursor appropriately if it isn't already
+     * positioned.
+     *
+     * In one scenario this function returns and needs to walk to the next record. In that case we
+     * set a boolean. We need to be careful that the entry flag isn't set so we don't re-enter
+     * search near.
+     *
+     * This positioning logic is at a valid place here. Moving it below __wt_cursor_func_init
```

Comment:
Try rewording to: "This logic must come before the call to __wt_cursor_func_init, as it will set the cbt active flag which changes the logic flow within the subsequent search near resulting in a segmentation fault. It is better to not set state prior to calling search near."